src/clr-angular/utils/focus/focus-on-view-init/focus-on-view-init.spec.ts

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright (c) 2016-2019 VMware, Inc. All Rights Reserved.
+ * This software is released under MIT license.
+ * The full license information can be found in LICENSE in the root directory of this project.
+ */
+
+import { Component, ViewChild, ElementRef } from '@angular/core';
+import { TestBed, ComponentFixture } from '@angular/core/testing';
+
+import { ClrFocusOnViewInit } from './focus-on-view-init';
+import { ClrFocusOnViewInitModule } from './focus-on-view-init.module';
+
+@Component({
+  template: `
+    <button #dummyButton>dummy button</button>
+    <div *ngIf="displayNoExistingTabindex">
+      <h1 clrFocusOnViewInit>Title</h1>
+    </div>
+    <div *ngIf="displayExistingTabindex">
+      <h1 clrFocusOnViewInit tabindex="5">Title</h1>
+    </div>
+  `,
+})
+class TestComponent {
+  displayNoExistingTabindex = false;
+  displayExistingTabindex = false;
+  @ViewChild('dummyButton', { static: true })
+  buttonElRef: ElementRef;
+
+  @ViewChild(ClrFocusOnViewInit, { static: false, read: ElementRef })
+  focusOnItElRef: ElementRef;
+}
+
+let fixture: ComponentFixture<any>;
+let component: TestComponent;
+
+fdescribe('ClrFocusOnViewInit', () => {
+  beforeEach(() => {
+    TestBed.configureTestingModule({
+      imports: [ClrFocusOnViewInitModule],
+      declarations: [TestComponent],
+    });
+    fixture = TestBed.createComponent(TestComponent);
+    component = fixture.componentInstance;
+    fixture.detectChanges();
+  });
+
+  it('sets negative tabindex if no tabindex', () => {
+    component.displayNoExistingTabindex = true;
+    fixture.detectChanges();
+    expect(component.focusOnItElRef.nativeElement.getAttribute('tabindex')).toBe('-1');
+  });
+
+  it('should use existing tabindex', () => {
+    component.displayExistingTabindex = true;
+    fixture.detectChanges();
+    expect(component.focusOnItElRef.nativeElement.getAttribute('tabindex')).toBe('5');
+  });
+
+  it('sets focus on the directive element on view intialization', () => {
+    component.buttonElRef.nativeElement.focus();
+    expect(document.activeElement).toBe(component.buttonElRef.nativeElement);
+    component.displayNoExistingTabindex = true;
+    fixture.detectChanges();
+    expect(document.activeElement).toBe(component.focusOnItElRef.nativeElement);
+  });
+
+  it('removes set tabindex attribute after focus shifts another element', () => {
+    component.displayNoExistingTabindex = true;
+    fixture.detectChanges();
+    expect(component.focusOnItElRef.nativeElement.getAttribute('tabindex')).toBe('-1');
+    component.buttonElRef.nativeElement.focus();
+    fixture.detectChanges();
+    expect(component.focusOnItElRef.nativeElement.getAttribute('tabindex')).toBeNull();
+  });
+
+  it('keeps existing attribute after focus shifts another element', () => {
```

Comment:
Maybe change the name/description? This seems to be testing the custom tabindex behavior, after focus shifts to anouther element.